File: Power analysis.py
import statsmodels.api as sm
from statsmodels.stats.power import FTestAnovaPower

# Set your parameters
effect_size = 0.5  # Cohen's d for a medium effect size
#effect_size = 0.8  # Cohen's d for a large effect size
alpha = 0.05      # Significance level
power = 0.80      # Desired statistical power
num_groups = 3    # Number of groups (sentiment versions)

# Create an instance of the FTestAnovaPower class
power_analysis = FTestAnovaPower()

# Perform the power analysis
sample_size = power_analysis.solve_power(effect_size=effect_size, alpha=alpha, nobs=None, power=power, k_groups=num_groups)

print(f"Required Sample Size per Group: {sample_size:.2f}")

# For medium effect size: n = 41.69 per group = 126 total
# For large effect size: n = 18.41 per group = 57 total

File: Power_analysis.R

library(pwr)

# Set parameters
alpha <- 0.05  # Significance level
power <- 0.80  # Desired power
#effect_size <- 0.2  # Cohen's d effect size (small effect)
effect_size <- 0.5  # Cohen's d effect size (medium effect)
#effect_size <- 0.8  # Cohen's d effect size (large effect)


# Number of groups (sentiments)
num_groups <- 3

# Perform power analysis for one-way ANOVA
result <- pwr.anova.test(k = num_groups, f = effect_size, sig.level = alpha, power = power)

# Display the result
result
"""
Small effect size:
  n = 81.296 = 82 participants in each group 
  246 in total.

Medium effect size:
  n = 13.895 = 14 in each group
  42 in total.
  
Large effect size:
  n = 6.136 = 7 in each group
  21 in total.
"""